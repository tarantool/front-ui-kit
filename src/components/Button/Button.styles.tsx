import { css } from '@emotion/css';

import { baseFontFamily, colors } from '../../variables';

export const styles = {
  button: css`
    white-space: nowrap;
    border: none;
    border-radius: 4px;
    box-sizing: border-box;
    font-family: ${baseFontFamily};
    font-size: 14px;
    line-height: 22px;
    transition-timing-function: ease-in-out;
    transition-duration: 0.07s;
    transition-property: background-color, color;
    outline: none;
    cursor: pointer;
    -webkit-font-smoothing: antialiased;

    &:hover,
    &:focus,
    &:active {
      transition-property: background-color, color, box-shadow;
    }

    &:disabled,
    &:disabled:active,
    &:disabled:hover {
      cursor: default;
    }
  `,

  icon: css`
    width: 16px;
    height: 16px;
    margin-bottom: 2px;
  `,

  iconMargin: css`
    margin-right: 8px;
  `,

  iconRightMargin: css`
    margin-left: 8px;
  `,

  loading: css`
    position: relative;
    color: rgba(0, 0, 0, 0);
    transition: none;

    &:hover,
    &:focus,
    &:active {
      cursor: default;
      color: rgba(0, 0, 0, 0);
    }

    & > *,
    &:hover > *,
    &:active > *,
    &:focus > * {
      visibility: hidden;
    }

    & > *:last-child {
      visibility: visible;
    }
  `,

  loadingWrap: css`
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
    display: flex;
    justify-content: center;
    align-items: center;
  `,

  l: css`
    padding: 9px 20px;
  `,

  m: css`
    padding: 5px 16px;
  `,

  s: css`
    padding: 1px 16px;
  `,

  xs: css`
    padding: 1px 9px;
    line-height: 18px;
    font-size: 12px;
  `,

  iconicL: css`
    padding: 9px 12px;
  `,

  iconicM: css`
    padding: 5px 8px;
  `,

  iconicS: css`
    padding: 1px 4px;
  `,
};

export const intentStyles = {
  base: css`
    background-color: white;
    color: ${colors.dark65};
    box-shadow: inset 0 0 0 1px ${colors.baseBg};

    &:disabled,
    &:disabled:active,
    &:disabled:hover {
      color: ${colors.intentBase};
      background-color: ${colors.intentBaseActive};
      box-shadow: inset 0 0 0 1px ${colors.intentBase};
    }

    & svg {
      fill: ${colors.dark65};
    }

    &:disabled svg {
      fill: ${colors.intentBase};
    }
  `,

  dark: css`
    background-color: ${colors.dark40};
    color: #ffffff;

    &:disabled,
    &:disabled:active,
    &:disabled:hover {
      color: ${colors.intentBase};
      background-color: ${colors.intentBaseActive};
    }

    & svg {
      fill: #ffffff;
    }

    &:disabled svg {
      fill: ${colors.intentBase};
    }
  `,

  primary: css`
    background-color: ${colors.intentPrimary};
    color: #ffffff;

    &:disabled,
    &:disabled:active,
    &:disabled:hover {
      background-color: ${colors.intentPrimaryDisabled};
      color: #ffffff;
    }

    & svg {
      fill: #ffffff;
    }

    &:disabled svg {
      fill: #ffffff;
    }
  `,

  secondary: css`
    background-color: ${colors.dark10};
    color: ${colors.dark65};

    &:disabled,
    &:disabled:active,
    &:disabled:hover {
      background-color: ${colors.intentBaseActive};
      color: ${colors.intentBase};
      box-shadow: inset 0 0 0 1px ${colors.intentBase};
    }

    & svg {
      fill: ${colors.dark65};
    }

    &:disabled svg {
      fill: ${colors.intentBase};
    }
  `,

  plain: css`
    background-color: transparent;
    color: ${colors.dark65};

    &:disabled,
    &:disabled:active,
    &:disabled:hover {
      background-color: ${colors.intentBaseActive};
      color: ${colors.intentBase};
    }

    & svg {
      fill: ${colors.dark65};
    }

    &:disabled svg {
      fill: ${colors.intentBase};
    }
  `,
};

export const intentActiveStyles = {
  base: css`
    &:focus,
    &:hover {
      background-color: ${colors.dark10};
      box-shadow: none;
    }

    &:active {
      background-color: ${colors.dark10};
      box-shadow: inset 0 4px 0 rgba(4, 11, 29, 0.1);
    }
  `,

  dark: css`
    &:focus,
    &:hover {
      background-color: ${colors.dark60};
    }

    &:active {
      background-color: ${colors.dark60};
      box-shadow: inset 0 4px 0 rgba(4, 11, 29, 0.1);
    }
  `,

  primary: css`
    &:focus,
    &:hover {
      background: ${colors.intentPrimaryActive};
    }

    &:active {
      background: ${colors.intentPrimaryActive};
      box-shadow: inset 0 4px 0 rgba(4, 11, 29, 0.1);
    }
  `,

  secondary: css`
    &:focus,
    &:hover {
      background-color: ${colors.dark25};
    }

    &:active {
      background-color: ${colors.dark25};
      box-shadow: inset 0 4px 0 rgba(4, 11, 29, 0.1);
    }
  `,

  plain: css`
    &:focus,
    &:hover {
      background-color: ${colors.dark10};
    }
  `,
};
