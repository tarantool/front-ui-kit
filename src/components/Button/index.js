// @flow
import * as React from 'react';
import { css, cx } from 'emotion'

const styles = {
  btn: css`
    border: solid 1px gray;
  `
}

type ButtonProps = {
  className?: string,
  children?: React.Node,
  onClick?: (MouseEvent) => void
};

export default ({ children, className, ...props }: ButtonProps) => (
  <button className={cx(className, styles.btn)} {...props}>{children}</button>
);
