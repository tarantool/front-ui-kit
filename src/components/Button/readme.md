Button example:

```js
import { css, cx } from 'emotion';
import {
  ControlsPanel,
  FormField,
  IconBucket,
  IconChevron,
  Switcher,
  Text
} from '../../index';

const styles = {
  wrapper: css`
    padding: 10px;
  `,
  bg: css`
    background-color: #f0f2f5;
  `
};

const label = 'Button';
const sizes = ['l', 'm', 's', 'xs'];
const intentions = ['primary', 'secondary', 'base', 'plain', 'dark'];

const DropdownControlIcon = props => <IconChevron direction='down' {...props} />;

initialState = {
  buttonText: true,
  darkBg: false,
  loading: false,
  leftIcon: false,
  rightIcon: false
};

const toggleBg = () => setState({ darkBg: !state.darkBg });
const toggleText = () => setState({
  buttonText: !state.buttonText,
  leftIcon: state.buttonText ? true : state.leftIcon
});
const toggleLeftIcon = () => setState({
  leftIcon: !state.leftIcon,
  buttonText: state.leftIcon ? true : state.buttonText
});
const toggleLoading = () => setState({ loading: !state.loading });
const toggleRightIcon = () => setState({ rightIcon: !state.rightIcon });

<div className={cx(styles.wrapper, { [styles.bg]: state.darkBg })}>
  <ControlsPanel
    controls={[
      <Switcher onChange={toggleText} checked={state.buttonText}>Button text</Switcher>,
      <Switcher onChange={toggleLoading} checked={state.loading}>Loading state</Switcher>,
      <Switcher onChange={toggleLeftIcon} checked={state.leftIcon}>Icon</Switcher>,
      <Switcher onChange={toggleRightIcon} checked={state.rightIcon}>Right icon</Switcher>,
      <Switcher onChange={toggleBg} checked={state.darkBg}>Background</Switcher>
    ]}
  />

  {intentions.map(intent => (
    <FormField label={intent} verticalSort={false} columns={31}>
      {sizes.reduce(
        (acc, size) => (
          [
            ...acc,
            <Button
              icon={state.leftIcon && IconBucket}
              iconRight={state.rightIcon && DropdownControlIcon}
              intent={intent}
              loading={state.loading}
              size={size}
              text={state.buttonText && label}
              title='Click me right meow!'
            />,
            <Button
              icon={state.leftIcon && IconBucket}
              iconRight={state.rightIcon && DropdownControlIcon}
              intent={intent}
              loading={state.loading}
              size={size}
              text={state.buttonText && label}
              title='Click me right meow!'
              disabled
            />
          ]
        ),
        []
      )}
    </FormField>
  ))}
  {(state.leftIcon || state.rightIcon) && (
    <Text tag='p'>The style guide doesn't recommend to use icons in 'xs' sized buttons.</Text>
  )}
</div>
```
