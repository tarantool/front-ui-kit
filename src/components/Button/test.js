import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from './index';
import { IconOk } from '../Icon';

describe('Simple buttons renders correctly', () => {
  it('idle state', () => {
    const tree = renderer.create(
      <>
        <Button title='Click me right meow!'>Click me</Button>
        <Button title='Click me right meow!' disabled>Click me</Button>
        <Button intent='primary'>Click me</Button>
        <Button intent='primary' disabled>Click me</Button>
        <Button intent='secondary'>Click me</Button>
        <Button intent='secondary' disabled>Click me</Button>
        <Button intent='plain'>Click me</Button>
        <Button intent='plain' disabled>Click me</Button>
        <Button intent='dark'>Click me</Button>
        <Button intent='dark' disabled>Click me</Button>

        <Button size='s' title='Click me right meow!'>Click me</Button>
        <Button size='s' title='Click me right meow!' disabled>Click me</Button>
        <Button size='s' intent='primary'>Click me</Button>
        <Button size='s' intent='primary' disabled>Click me</Button>
        <Button size='s' intent='secondary'>Click me</Button>
        <Button size='s' intent='secondary' disabled>Click me</Button>
        <Button size='s' intent='plain'>Click me</Button>
        <Button size='s' intent='plain' disabled>Click me</Button>
        <Button size='s' intent='dark'>Click me</Button>
        <Button size='s' intent='dark' disabled>Click me</Button>

        <Button size='xs' title='Click me right meow!'>Click me</Button>
        <Button size='xs' title='Click me right meow!' disabled>Click me</Button>
        <Button size='xs' intent='primary'>Click me</Button>
        <Button size='xs' intent='primary' disabled>Click me</Button>
        <Button size='xs' intent='secondary'>Click me</Button>
        <Button size='xs' intent='secondary' disabled>Click me</Button>
        <Button size='xs' intent='plain'>Click me</Button>
        <Button size='xs' intent='plain' disabled>Click me</Button>
        <Button size='xs' intent='dark'>Click me</Button>
        <Button size='xs' intent='dark' disabled>Click me</Button>
      </>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('loading state', () => {
    const tree = renderer.create(
      <>
        <Button loading title='Click me right meow!'>Click me</Button>
        <Button loading title='Click me right meow!' disabled>Click me</Button>
        <Button loading intent='primary'>Click me</Button>
        <Button loading intent='primary' disabled>Click me</Button>
        <Button loading intent='secondary'>Click me</Button>
        <Button loading intent='secondary' disabled>Click me</Button>
        <Button loading intent='plain'>Click me</Button>
        <Button loading intent='plain' disabled>Click me</Button>
        <Button loading intent='dark'>Click me</Button>
        <Button loading intent='dark' disabled>Click me</Button>

        <Button loading size='s' title='Click me right meow!'>Click me</Button>
        <Button loading size='s' title='Click me right meow!' disabled>Click me</Button>
        <Button loading size='s' intent='primary'>Click me</Button>
        <Button loading size='s' intent='primary' disabled>Click me</Button>
        <Button loading size='s' intent='secondary'>Click me</Button>
        <Button loading size='s' intent='secondary' disabled>Click me</Button>
        <Button loading size='s' intent='plain'>Click me</Button>
        <Button loading size='s' intent='plain' disabled>Click me</Button>
        <Button loading size='s' intent='dark'>Click me</Button>
        <Button loading size='s' intent='dark' disabled>Click me</Button>

        <Button loading size='xs' title='Click me right meow!'>Click me</Button>
        <Button loading size='xs' title='Click me right meow!' disabled>Click me</Button>
        <Button loading size='xs' intent='primary'>Click me</Button>
        <Button loading size='xs' intent='primary' disabled>Click me</Button>
        <Button loading size='xs' intent='secondary'>Click me</Button>
        <Button loading size='xs' intent='secondary' disabled>Click me</Button>
        <Button loading size='xs' intent='plain'>Click me</Button>
        <Button loading size='xs' intent='plain' disabled>Click me</Button>
        <Button loading size='xs' intent='dark'>Click me</Button>
        <Button loading size='xs' intent='dark' disabled>Click me</Button>
      </>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('Buttons with icons renders correctly', () => {
  it('idle state', () => {
    const tree = renderer.create(
      <>
        <Button icon={IconOk} title='Click me right meow!'>Click me</Button>
        <Button icon={IconOk} title='Click me right meow!' disabled>Click me</Button>
        <Button icon={IconOk} intent='primary'>Click me</Button>
        <Button icon={IconOk} intent='primary' disabled>Click me</Button>
        <Button icon={IconOk} intent='secondary'>Click me</Button>
        <Button icon={IconOk} intent='secondary' disabled>Click me</Button>
        <Button icon={IconOk} intent='plain'>Click me</Button>
        <Button icon={IconOk} intent='plain' disabled>Click me</Button>
        <Button icon={IconOk} intent='dark'>Click me</Button>
        <Button icon={IconOk} intent='dark' disabled>Click me</Button>

        <Button icon={IconOk} size='s' title='Click me right meow!'>Click me</Button>
        <Button icon={IconOk} size='s' title='Click me right meow!' disabled>Click me</Button>
        <Button icon={IconOk} size='s' intent='primary'>Click me</Button>
        <Button icon={IconOk} size='s' intent='primary' disabled>Click me</Button>
        <Button icon={IconOk} size='s' intent='secondary'>Click me</Button>
        <Button icon={IconOk} size='s' intent='secondary' disabled>Click me</Button>
        <Button icon={IconOk} size='s' intent='plain'>Click me</Button>
        <Button icon={IconOk} size='s' intent='plain' disabled>Click me</Button>
        <Button icon={IconOk} size='s' intent='dark'>Click me</Button>
        <Button icon={IconOk} size='s' intent='dark' disabled>Click me</Button>

        <Button iconRight={IconOk} title='Click me right meow!'>Click me</Button>
        <Button iconRight={IconOk} title='Click me right meow!' disabled>Click me</Button>
        <Button iconRight={IconOk} intent='primary'>Click me</Button>
        <Button iconRight={IconOk} intent='primary' disabled>Click me</Button>
        <Button iconRight={IconOk} intent='secondary'>Click me</Button>
        <Button iconRight={IconOk} intent='secondary' disabled>Click me</Button>
        <Button iconRight={IconOk} intent='plain'>Click me</Button>
        <Button iconRight={IconOk} intent='plain' disabled>Click me</Button>
        <Button iconRight={IconOk} intent='dark'>Click me</Button>
        <Button iconRight={IconOk} intent='dark' disabled>Click me</Button>

        <Button iconRight={IconOk} size='s' title='Click me right meow!'>Click me</Button>
        <Button iconRight={IconOk} size='s' title='Click me right meow!' disabled>Click me</Button>
        <Button iconRight={IconOk} size='s' intent='primary'>Click me</Button>
        <Button iconRight={IconOk} size='s' intent='primary' disabled>Click me</Button>
        <Button iconRight={IconOk} size='s' intent='secondary'>Click me</Button>
        <Button iconRight={IconOk} size='s' intent='secondary' disabled>Click me</Button>
        <Button iconRight={IconOk} size='s' intent='plain'>Click me</Button>
        <Button iconRight={IconOk} size='s' intent='plain' disabled>Click me</Button>
        <Button iconRight={IconOk} size='s' intent='dark'>Click me</Button>
        <Button iconRight={IconOk} size='s' intent='dark' disabled>Click me</Button>

        <Button icon={IconOk} iconRight={IconOk} title='Click me right meow!'>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} title='Click me right meow!' disabled>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} intent='primary'>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} intent='primary' disabled>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} intent='secondary'>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} intent='secondary' disabled>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} intent='plain'>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} intent='plain' disabled>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} intent='dark'>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} intent='dark' disabled>Click me</Button>

        <Button icon={IconOk} iconRight={IconOk} size='s' title='Click me right meow!'>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} size='s' title='Click me right meow!' disabled>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} size='s' intent='primary'>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} size='s' intent='primary' disabled>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} size='s' intent='secondary'>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} size='s' intent='secondary' disabled>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} size='s' intent='plain'>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} size='s' intent='plain' disabled>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} size='s' intent='dark'>Click me</Button>
        <Button icon={IconOk} iconRight={IconOk} size='s' intent='dark' disabled>Click me</Button>
      </>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('loading state', () => {
    const tree = renderer.create(
      <>
        <Button icon={IconOk} loading title='Click me right meow!'>Click me</Button>
        <Button icon={IconOk} loading title='Click me right meow!' disabled>Click me</Button>
        <Button icon={IconOk} loading intent='primary'>Click me</Button>
        <Button icon={IconOk} loading intent='primary' disabled>Click me</Button>
        <Button icon={IconOk} loading intent='secondary'>Click me</Button>
        <Button icon={IconOk} loading intent='secondary' disabled>Click me</Button>
        <Button icon={IconOk} loading intent='plain'>Click me</Button>
        <Button icon={IconOk} loading intent='plain' disabled>Click me</Button>
        <Button icon={IconOk} loading intent='dark'>Click me</Button>
        <Button icon={IconOk} loading intent='dark' disabled>Click me</Button>

        <Button icon={IconOk} loading size='s' title='Click me right meow!'>Click me</Button>
        <Button icon={IconOk} loading size='s' title='Click me right meow!' disabled>Click me</Button>
        <Button icon={IconOk} loading size='s' intent='primary'>Click me</Button>
        <Button icon={IconOk} loading size='s' intent='primary' disabled>Click me</Button>
        <Button icon={IconOk} loading size='s' intent='secondary'>Click me</Button>
        <Button icon={IconOk} loading size='s' intent='secondary' disabled>Click me</Button>
        <Button icon={IconOk} loading size='s' intent='plain'>Click me</Button>
        <Button icon={IconOk} loading size='s' intent='plain' disabled>Click me</Button>
        <Button icon={IconOk} loading size='s' intent='dark'>Click me</Button>
        <Button icon={IconOk} loading size='s' intent='dark' disabled>Click me</Button>
      </>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
