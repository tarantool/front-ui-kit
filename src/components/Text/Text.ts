import { MouseEvent, ReactNode, createElement, forwardRef } from 'react';
import { cx } from '@emotion/css';

import { textStyles } from './Text.styles';

export { textStyles };

export type TextStyleVariants = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'p' | 'basic' | 'code';

export interface TextProps {
  className?: string;
  htmlFor?: string;
  children?: ReactNode;
  tag?: string;
  href?: string;
  upperCase?: boolean;
  noCase?: boolean;
  title?: string;
  variant?: TextStyleVariants;
  onClick?: (event: MouseEvent<HTMLElement>) => void;
  // colorVariant?: 'inherit' | 'primary' | 'secondary' | 'disabled' | 'white' | 'error',
}

export const Text = forwardRef<HTMLElement, TextProps>(
  ({ className, children, tag, upperCase, noCase, variant = 'basic', ...props }, ref) => {
    const Element = tag || (variant === 'basic' ? 'span' : variant);

    return createElement(
      Element,
      {
        ...props,
        className: cx(
          textStyles[variant],
          {
            [textStyles.upperCase]: upperCase,
            [textStyles.noCase]: noCase,
          },
          className
        ),
        ref,
      },
      children
    );
  }
);
