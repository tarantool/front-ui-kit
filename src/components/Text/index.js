// @flow
import * as React from 'react'
import { css, cx } from 'emotion'

export const textStyles = {
  h1: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 24px;
    line-height: 38px;
    font-weight: 600;
    letter-spacing: 0.48px;
    text-transform: uppercase;
    color: #000000;
  `,
  h2: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 20px;
    line-height: 28px;
    font-weight: 600;
    letter-spacing: 0.4px;
    text-transform: uppercase;
    color: #000000;
  `,
  h3: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 16px;
    line-height: 24px;
    font-weight: 600;
    letter-spacing: 0.32px;
    text-transform: uppercase;
    color: #cf1322;
  `,
  h4: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 16px;
    line-height: 24px;
    font-weight: 600;
    letter-spacing: 0.32px;
    color: #000000;
  `,
  h5: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 12px;
    line-height: 22px;
    letter-spacing: 0.24px;
    color: #000000;
  `,
  p: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 12px;
    line-height: 20px;
    letter-spacing: 0.28px;
    color: #000000;
  `,
  code: css`
    margin: 0;
    font-family: 'Monaco', monospace;
    font-size: 14px;
    line-height: 22px;
    color: #000000;
    b {
      font-weight: 600;
    }
  `,
  basic: css`
    margin: 0;
    font-family: 'Open Sans', sans-serif;
    font-size: 14px;
    line-height: 22px;
    color: #000000;
    b {
      font-weight: 600;
    }
  `,
  upperCase: css`
    text-transform: uppercase;
  `,
  noCase: css`
    text-transform: none;
  `
};

type TextProps = {
  className?: string,
  children?: React.Node,
  tag?: string,
  upperCase?: boolean,
  noCase?: boolean,
  title?: string,
  variant?:
    |'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'p' | 'basic' | 'code',
  // colorVariant?: 'inherit' | 'primary' | 'secondary' | 'disabled' | 'white' | 'error',
};

export const Text = ({
  className,
  children,
  tag,
  upperCase,
  noCase,
  title,
  variant = 'basic'
}:
TextProps) => {
  const Element = tag || (variant === 'basic' ? 'span' : variant);

  return (
    <Element
      className={cx(
        textStyles[variant],
        {
          [textStyles.upperCase]: upperCase,
          [textStyles.noCase]: noCase
        },
        className
      )}
      title={title}
    >
      {children}
    </Element>
  );
};
