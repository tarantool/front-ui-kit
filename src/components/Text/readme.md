```
<Text variant='h1'>Heading 1</Text>
<Text variant='h2'>Heading 2</Text>
<Text variant='h3'>Heading 3</Text>
<Text variant='h4'>Heading 4</Text>
<Text variant='h5'>Heading 5</Text>
<Text variant='p'>paragraph</Text>
<Text>basic</Text>
<br />
<Text variant='h3' tag='span'>H3 styled span</Text>
<Text variant='h4' tag='span'>H4 styled span</Text>
```
