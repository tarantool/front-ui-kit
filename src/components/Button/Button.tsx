
import React, { ReactNode, MouseEvent, forwardRef, ComponentType } from 'react';
import { cx } from '@emotion/css';

import { IconSpinner } from '../Icon';
import { styles, intentStyles, intentActiveStyles } from './Button.styles';

export interface ButtonIconProps {
  className?: string;
}

export interface ButtonProps {
  autoFocus?: boolean,
  className?: string,
  children?: ReactNode,
  disabled?: boolean,
  icon?: ComponentType<ButtonIconProps>,
  iconRight?: ComponentType<ButtonIconProps>,
  intent?: 'primary' | 'secondary' | 'base' | 'plain' | 'dark',
  onClick?: (e: MouseEvent<HTMLButtonElement>) => void,
  loading?: boolean,
  size?: 'l' | 's' | 'xs' | 'm',
  text?: string,
  title?: string,
  type?: 'button' | 'submit'
}

export const Button = forwardRef<HTMLButtonElement, ButtonProps>((
  {
    autoFocus,
    className,
    children,
    disabled,
    icon: Icon,
    iconRight: IconRight,
    intent = 'base',
    onClick,
    loading,
    size = 'm',
    text,
    title,
    type = 'button',
    ...props
  }: ButtonProps,
  ref
) => {
  const isOnlyIcon = Icon && !children && !text;

  const content: JSX.Element[] = [];

  if (Icon) {
    content.push(
      <Icon
        key="Icon"
        className={cx(
          styles.icon,
          {
            [styles.iconMargin]: !isOnlyIcon
          }
        )}
      />
    )
  }

  content.push(<React.Fragment key="content">{children || text}</React.Fragment>)

  if (IconRight && !isOnlyIcon) {
    content.push(
      <IconRight
        key="IconRight"
        className={cx(
          styles.icon,
          styles.iconRightMargin
        )}
      />
    )
  }

  if (loading && !disabled) {
    content.push(
      <div key="loading" className={styles.loadingWrap}>
        <IconSpinner />
      </div>
    );
  }

  return (
    <button
      {...props}
      autoFocus={autoFocus}
      className={cx(
        styles.button,
        {
          [styles.iconicL]: isOnlyIcon && size === 'l',
          [styles.iconicM]: isOnlyIcon && size === 'm',
          [styles.iconicS]: isOnlyIcon && size === 's',
          [styles.l]: !isOnlyIcon && size === 'l',
          [styles.m]: !isOnlyIcon && size === 'm',
          [styles.s]: !isOnlyIcon && size === 's',
          [styles.xs]: size === 'xs',
          [intentActiveStyles[intent]]: !loading && !disabled
        },
        intentStyles[intent],
        {
          [styles.loading]: loading && !disabled
        },
        className
      )}
      disabled={disabled}
      onClick={onClick}
      title={title}
      type={type}
      ref={ref}
    >
      {content}
    </button>
  );
});
