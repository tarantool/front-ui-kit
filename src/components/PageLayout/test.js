import React from 'react';
import renderer from 'react-test-renderer';
import { Button } from '../Button';
import { Text } from '../Text';
import { PageLayout } from './index';

it('PageLayout renders succesfully', () => {
  const tree = renderer.create(
    <>
      <PageLayout
        heading='Cluster'
        headingContent={<Text>Heading content</Text>}
      />
      <PageLayout
        heading='Cluster'
        controls={[
          <Button text='Details' />,
          <Button text='Issues' />
        ]}
        wide
      />
    </>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
