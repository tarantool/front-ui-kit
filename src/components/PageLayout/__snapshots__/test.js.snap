// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageLayout renders succesfully 1`] = `
Array [
  <div
    className="css-1tgvd0c"
  >
    <div
      className="css-pyr5ve"
    >
      <h1
        className="css-1xisb9e"
      >
        Cluster
      </h1>
      <span
        className="css-1ggp6x0"
      >
        Heading content
      </span>
    </div>
  </div>,
  <div
    className="css-13d9vqf"
  >
    <div
      className="css-pyr5ve"
    >
      <h1
        className="css-1xisb9e"
      >
        Cluster
      </h1>
      <div
        className="css-70qvj9"
      >
        <div
          className="css-12ibiio"
        >
          <button
            className="css-1swuuzl"
            type="button"
          >
            Details
          </button>
        </div>
        <div
          className="css-12ibiio"
        >
          <button
            className="css-1swuuzl"
            type="button"
          >
            Issues
          </button>
        </div>
      </div>
    </div>
  </div>,
]
`;
